app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/UserPhotoFragmentTest.kt

```diff
@@ -0,0 +1,163 @@
+                Date()
+            ),User("0",100, "SuperUser69"), "Organique","Wowa je suis incroyable en train de ramasser cette couche usagée pour faire un selfie avec!"),
+            PhotoMetadata("1","Look at me cleaning!", ParcelableDate(Date()),User("0",100, "SuperUser69"), "Organique","Wowa je suis incroyable en train de ramasser cette couche usagée pour faire un selfie avec!"),
+            PhotoMetadata("1","Look at me cleaning!", ParcelableDate(Date()),User("0",100, "SuperUser69"), "Organique","Wowa je suis incroyable en train de ramasser cette couche usagée pour faire un selfie avec!"),
+            PhotoMetadata("1","Look at me cleaning!", ParcelableDate(Date()),User("0",100, "SuperUser69"), "Organique","Wowa je suis incroyable en train de ramasser cette couche usagée pour faire un selfie avec!")
+        )
+        photoList = photos
+
+        // Create a new instance of the fragment with test arguments.
+        fragment = UserPhotoFragment.newInstance(2, photos, true)
+    }
+
+    @Test
+    fun onCreateViewFeedWithValidViewReturnsView() {
+        val args = Bundle().apply {
+            putInt(ARG_COLUMN_COUNT, 1)
+            putParcelableArrayList(ARG_PHOTO_LIST, photoList.toCollection(ArrayList()))
+            putBoolean(ARG_IS_FEED, true)
+        }
+
+        // Launch fragment with arguments
+        val scenario = FragmentScenario.launchInContainer(UserPhotoFragment::class.java, args)
+        // Wait for the fragment to be created
+        scenario.onFragment { fragment ->
+            val recyclerView = fragment.view
+            assertNotNull(recyclerView)
+            assertTrue(recyclerView is RecyclerView)
+        }
+    }
+
+    @Test
+    fun onCreateViewWithValidViewReturnsView() {
+        val args = Bundle().apply {
+            putInt(ARG_COLUMN_COUNT, 1)
+            putParcelableArrayList(ARG_PHOTO_LIST, photoList.toCollection(ArrayList()))
+            putBoolean(ARG_IS_FEED, false)
+        }
+
+        // Launch fragment with arguments
+        val scenario = FragmentScenario.launchInContainer(UserPhotoFragment::class.java, args)
+        // Wait for the fragment to be created
+        scenario.onFragment { fragment ->
+            val recyclerView = fragment.view
+            assertNotNull(recyclerView)
+            assertTrue(recyclerView is RecyclerView)
+        }
+    }
+
+    @Test
+    fun onCreateViewWithListShowsList() {
+        // Set up the inflater and container to create a valid view.
+        val args = Bundle().apply {
+            putInt(ARG_COLUMN_COUNT, 1)
+            putParcelableArrayList(ARG_PHOTO_LIST, photoList.toCollection(ArrayList()))
+        }
+
+        // Launch fragment with arguments
+        val scenario = FragmentScenario.launchInContainer(UserPhotoFragment::class.java, args)
```

Comment:
Don't forget to close the scenario with :

`scenario.close()`

Closing the scenario ensures that all the resources used during the test are released, and the application is left in its original state before starting the next test.

Same for other tests.

Reply:
Oh yes thanks for the advice!